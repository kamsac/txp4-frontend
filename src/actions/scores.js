import dotenvConfiguration from '../dotenv-configuration';
import ScoresResourceMock from '../resources/scores/mock';
import ScoresResource from '../resources/scores';

export const REQUEST_SCORES = 'scores:requestScores';
export function requestScores() {
  return dispatch => {
    return dispatch(fetchScores());
  }
}

function fetchScores() {
  const resource = dotenvConfiguration.API_URL ? ScoresResource : ScoresResourceMock;

  return dispatch => {
    resource.getScores()
      .then((response) => {
        dispatch(receiveScores(response.data));
      });
  }
}

export const RECEIVE_SCORES = 'scores:fetchScoresSuccess';
export function receiveScores(scores) {
  return {
    type: RECEIVE_SCORES,
    payload: { scores },
  }
}
