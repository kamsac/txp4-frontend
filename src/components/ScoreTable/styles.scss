@import '../../variables';

$score-table-cell-padding-x: 1rem;
$score-table-cell-padding-y: 0.6rem;
$score-table-header-padding-x: $score-table-cell-padding-x;
$score-table-header-padding-y: 1rem;

.ScoreTable {
  border: $border-width solid $black;
  border-collapse: collapse;
  grid-area: ScoreTable;
  width: 100%;
}

.ScoreTable-HeaderRow {
  background-color: $black;
  color: $white;
}

.ScoreTable-Row {
  &:hover {
    background-color: $data-hover-color;
  }
}

.ScoreTable-HeaderCell {
  padding: $score-table-header-padding-y $score-table-header-padding-x;
}

.ScoreTable-Cell {
  padding: 0;
}

.ScoreTable-Cell--Nickname {
  text-decoration: none;
  text-align: left;
}

.ScoreTable-Cell--Points {
  text-decoration: none;
  text-align: right;
}

.ScoreTable-Cell-Anchor {
  color: $unset;
  text-decoration: none;
  display: block;
  padding: $score-table-cell-padding-y $score-table-cell-padding-x;
}
