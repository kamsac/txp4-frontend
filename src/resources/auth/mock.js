export const ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJsb2dpbiI6Imx5bWFrIiwibmljayI6IiRzJG8kYTYwbCRjODJ5JGZhM20kYzgyYSRhNjBrIn0.aaKzXEZP2tovhBAqPJYBdmbLzKjfdBsuX3FaB58JZvE';

const AuthResourceMock = {
  login() {
    return new Promise((resolve) => {
      const response = {
        data: {
          token: ACCESS_TOKEN,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default AuthResourceMock;
