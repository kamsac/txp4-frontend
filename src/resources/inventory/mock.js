const ITEMS = [
  {
    id: "a1a1a1",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "b2b2b2",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "c3c3c3",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "d4d4d4",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "e5e5e5",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "f6f6f6",
    name: "Duper engine",
    vendorId: 'aaa',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1.5,
      },
      {
        name: "acceleration",
        value: 3,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "f7f7f7",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "g8g8g8",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "h9h9h9",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "i0i0i0",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "j1j1j1",
    name: "Transmission",
    vendorId: 'ccc',
    equipRegion: "transmission",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.7,
      },
      {
        name: "acceleration",
        value: 1.5,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
];

const InventoryResourceMock = {
  getItems() {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem(item) {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(ITEMS), 100);
    });
  },
};

export default InventoryResourceMock;
