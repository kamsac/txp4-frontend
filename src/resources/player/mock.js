export const ITEMS = [
  {
    id: 'a1a1a1',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'b2b2b2',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'c3c3c3',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'd4d4d4',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'e5e5e5',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'f6f6f6',
    name: 'Duper engine',
    vendorId: 'aaa',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1.5,
      },
      {
        name: 'acceleration',
        value: 3,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'f7f7f7',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'g8g8g8',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'h9h9h9',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'i0i0i0',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'j1j1j1',
    name: 'Transmission',
    vendorId: 'ccc',
    equipRegion: 'transmission',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.7,
      },
      {
        name: 'acceleration',
        value: 1.5,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
];

export const PLAYER = {
  login: 'lymak',
  nick: '$s$o$a60l$c82y$fa3m$c82a$a60k',
  inventory: {
    items: ITEMS,
  },
};

const PlayerResourceMock = {
  getPlayer() {
    return new Promise((resolve) => {
      const response = {
        data: PLAYER,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  getInventory() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  unequipItem() {
    return new Promise((resolve) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default PlayerResourceMock;
