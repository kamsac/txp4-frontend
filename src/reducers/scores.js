import {RECEIVE_SCORES} from '../actions/scores';

const initialValue = [];

export function scoresReducer(state = initialValue, action) {
  switch(action.type) {
    case RECEIVE_SCORES:
      return [...action.payload.scores];
    default:
      return state;
  }
}
