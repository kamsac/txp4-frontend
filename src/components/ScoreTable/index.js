import React from 'react';
import './styles.scss';

export default class ScoreTable extends React.Component {
  constructor(props) {
    super(props);

    this.rowsLimit = props.rowsLimit || 10;
  }

  render() {
    const scoreTableRows = this.props.scores
      .slice(0, this.rowsLimit)
      .map((score) => {
        return (
          <tr className="ScoreTable-row" key={score.nickname}>
            <td className="ScoreTable-cell ScoreTable-cell--nickname">{score.player.nickname}</td>
            <td className="ScoreTable-cell ScoreTable-cell--points">{score.points}</td>
          </tr>
        );
      });

    return (
      <table className="ScoreTable">
        <thead>
          <tr className="ScoreTable-headerRow">
            <th className="ScoreTable-cell ScoreTable-cell--nickname">Nickname</th>
            <th className="ScoreTable-cell ScoreTable-cell--points">Points</th>
          </tr>
        </thead>
        <tbody>
          { scoreTableRows }
        </tbody>
      </table>
    );
  }
}
