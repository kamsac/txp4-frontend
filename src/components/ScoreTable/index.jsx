import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './styles.scss';
import { ScorePropTypesShape } from '../../prop-types';
import trackmaniaStyleParser from '../../trackmania-style-parser';

export default class ScoreTable extends React.Component {
  componentDidMount() {
    this.props.requestScores();
  }

  render() {
    const scoreTableRows = this.props.scores
      .slice(0, this.props.rowsLimit)
      .map(score => (
        <tr className="ScoreTable-Row" key={score.player.login}>
          <td className="ScoreTable-Cell ScoreTable-Cell--Nickname">
            <Link
              to={`/player/${score.player.login}`}
              className="ScoreTable-Cell-Anchor"
              dangerouslySetInnerHTML={{
                __html: trackmaniaStyleParser(score.player.nick),
              }}
            />
          </td>
          <td className="ScoreTable-Cell ScoreTable-Cell--Points">
            <Link
              to={`/player/${score.player.login}`}
              className="ScoreTable-Cell-Anchor"
            >
              {score.score}
            </Link>

          </td>
        </tr>
      ));

    return (
      <table className="ScoreTable">
        <thead>
          <tr className="ScoreTable-HeaderRow">
            <th className="ScoreTable-HeaderCell ScoreTable-Cell--Nickname">Nickname</th>
            <th className="ScoreTable-HeaderCell ScoreTable-Cell--Points">Points</th>
          </tr>
        </thead>
        <tbody>
          { scoreTableRows }
        </tbody>
      </table>
    );
  }
}

ScoreTable.propTypes = {
  scores: PropTypes.arrayOf(ScorePropTypesShape).isRequired,
  rowsLimit: PropTypes.number,
  requestScores: PropTypes.func.isRequired,
};

ScoreTable.defaultProps = {
  rowsLimit: 10,
};
