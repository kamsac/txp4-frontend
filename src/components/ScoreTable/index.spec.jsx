import React from 'react';
import { shallow } from 'enzyme';
import ScoreTable from './index';
import { SCORES } from '../../resources/scores/mock';

test('<ScoreTable>', () => {
  const component = shallow(<ScoreTable
    scores={SCORES}
    requestScores={jest.fn}
  />);
  expect(component).toMatchSnapshot();
});

test('should display specified numbers of rows', () => {
  const rowsLimit = 3;
  const component = shallow(<ScoreTable
    rowsLimit={rowsLimit}
    scores={SCORES}
    requestScores={jest.fn}
  />);
  const rows = component.find('tbody tr');
  expect(rows.length).toEqual(rowsLimit);
});
