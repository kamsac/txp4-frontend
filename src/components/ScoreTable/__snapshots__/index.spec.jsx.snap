// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoreTable> 1`] = `
<table
  className="ScoreTable"
>
  <thead>
    <tr
      className="ScoreTable-HeaderRow"
    >
      <th
        className="ScoreTable-HeaderCell ScoreTable-Cell--Nickname"
      >
        Nickname
      </th>
      <th
        className="ScoreTable-HeaderCell ScoreTable-Cell--Points"
      >
        Points
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="ScoreTable-Row"
      key="lymak"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"font-weight:bold;color:#a60;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">l</span><span style=\\"font-weight:bold;color:#c82;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">y</span><span style=\\"font-weight:bold;color:#fa3;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">m</span><span style=\\"font-weight:bold;color:#c82;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">a</span><span style=\\"font-weight:bold;color:#a60;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);\\">k</span>",
            }
          }
          replace={false}
          to="/player/lymak"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/lymak"
        >
          4242
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="kibes"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"color:#fff;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);letter-spacing:.1em;font-size:105%;\\">kibes</span>",
            }
          }
          replace={false}
          to="/player/kibes"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/kibes"
        >
          3232
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="tomek"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"font-weight:bold;color:#08f;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);letter-spacing:.1em;font-size:105%;\\">tom</span><span style=\\"font-weight:bold;color:#fff;text-shadow:1px 1px 1px rgba(0, 0, 0, 0.5);letter-spacing:.1em;font-size:105%;\\">ek</span>",
            }
          }
          replace={false}
          to="/player/tomek"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/tomek"
        >
          2416
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="omed"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"color:#f80;\\">omed</span>",
            }
          }
          replace={false}
          to="/player/omed"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/omed"
        >
          2000
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="tar"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"color:#f00;\\">tar</span>",
            }
          }
          replace={false}
          to="/player/tar"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/tar"
        >
          1960
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="oiz"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"font-weight:bold;color:#000;letter-spacing:.1em;font-size:105%;\\">oiz</span>",
            }
          }
          replace={false}
          to="/player/oiz"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/oiz"
        >
          680
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="xetrow"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"color:#484;letter-spacing:.1em;font-size:105%;\\">xetrow</span>",
            }
          }
          replace={false}
          to="/player/xetrow"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/xetrow"
        >
          403
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="ginok"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"color:#880;letter-spacing:-.1em;font-size:95%;\\">ginok</span>",
            }
          }
          replace={false}
          to="/player/ginok"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/ginok"
        >
          30
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="llif"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"color:#400;letter-spacing:.1em;font-size:105%;\\">llif</span>",
            }
          }
          replace={false}
          to="/player/llif"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/llif"
        >
          20
        </Link>
      </td>
    </tr>
    <tr
      className="ScoreTable-Row"
      key="ida"
    >
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Nickname"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span style=\\"color:#f00;\\">ida</span>",
            }
          }
          replace={false}
          to="/player/ida"
        />
      </td>
      <td
        className="ScoreTable-Cell ScoreTable-Cell--Points"
      >
        <Link
          className="ScoreTable-Cell-Anchor"
          replace={false}
          to="/player/ida"
        >
          10
        </Link>
      </td>
    </tr>
  </tbody>
</table>
`;
